### PR TITLE
default_collate: Support regular tuples as well as namedtuple

### PR DESCRIPTION
**Pull request by cannon**

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
@SsnL Appreciate it if you could review this PR.

**Comment by SsnL** [comment]
I'm not against this. This is sort of BC breaking, but I don't think it matters much. But this deserves a test.

**Review by ssnl — CHANGES_REQUESTED** [review]
thanks! but this needs a test.

File: torch/utils/data/_utils/collate.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def default_collate(batch):
         return batch
     elif isinstance(batch[0], container_abcs.Mapping):
         return {key: default_collate([d[key] for d in batch]) for key in batch[0]}
-    elif isinstance(batch[0], tuple) and hasattr(batch[0], '_fields'):  # namedtuple
+    elif isinstance(batch[0], tuple):
         return type(batch[0])(*(default_collate(samples) for samples in zip(*batch)))
```
Comment (SsnL) [inline torch/utils/data/_utils/collate.py]: oh and the tuple ctor can not be called this way, so this should be another elif.

**Pull request closed without merging**